config.go

```diff
@@ -28,6 +29,16 @@ type Env struct {
 	Value *string `yaml:"value"`
 }
 
+type Options struct {
+	ScaleMin              *int    `yaml:"scale-min,omitempty"`
+	ScaleMax              *int    `yaml:"scale-max,omitempty"`
+	ScaleInit             *int    `yaml:"scale-init,omitempty"`
+	AutoscaleWindow       *string `yaml:"autoscale-window,omitempty"`
+	ConcurrencyLimit      *int64  `yaml:"concurrency-limit,omitempty"`
+	ConcurrencyTarget     *int    `yaml:"concurrency-target,omitempty"`
+	ConcurrencyUtilizatin *int    `yaml:"concurrency-utilization,omitempty"`
```

Comment:
I am not sure whether we should use the same notation as it is used in `kn` ie. dashes in the name or whether we should use CamelCase notation that's being used in the rest of the file...

Reply:
tl;dr  I see no compelling reason to maintain exact equivalence, and lean towards CamelCase at a minimum.

My first thought on reading this was much the same:  Glad we have it; a bit less enthusiastic about the naming.  I would even go a little further... Following is perhaps overly opinionated, but why not share!

I am sure that many would encourage us to maintain homogeneity between the two.  However, I think this actually leads to a couple challenges:  First, it encourages a false equivalence between the systems, causing confusion by users and developers alike; especially when we do deviate.  Second, we risk grandfathering in naming which, with the benefit of hindsight, could be easily improved or made to fit our use case a bit better.  I see the argument for exact equivalence as valid, but weaker than either:  "it's easier to remember if you're familiar with both systems".

We have a golden and ephemeral opportunity to lay down consistent, coherent naming now, before we are restrained by the iron yoke backwards compatibility.  I say we take it whenever the change is a clear improvement and not just bike-shedding.

For instance, would anyone argue that this is clearer semantics?
```
scaling:
  initial: 2
  min: 1
  max:  10
  window: 10s
concurrency:
  target: 75
  utilization: 75
  limit: 100
```
(This is of course just a conversational example more than an actual suggestion; illustrating a point.  I like the practical original suggestion of adopting CamelCase just fine)

Reply:
I actually like your "conversational example" a bit more than using an `options` keyword. My first thought upon seeing the PR was, "why an `options` key" instead of just having `scale-min`, for example, at the top level? I suppose there is benefit in the fact that, as new flags/options are added they can all just be stuffed into one `options` bag instead of being scattered across the file. I do agree, however, that your example, @lkingland is more readable/understandable. What about something like this (also a conversational example, but maybe also a suggestion :shrug:)

```yaml
options:
  scaling:
    initial: 2
    min: 1
    max:  10
    window: 10s
  concurrency:
    target: 75
    utilization: 75
    limit: 100
```

Reply:
Yeah, I created the `options` exactly for this, to not have all the settings scattered across the func.yaml file.

I like this suggestion, will refactor the config this way, thanks a lot for the input.